front/src/applications/operationalStudies/hooks.ts

```diff
@@ -88,20 +108,63 @@ export const useSetupItineraryForTrainUpdate = (
               const stepsCoordinates = pathfindingResult.path_items_positions.map((position) =>
                 getPointCoordinates(geometry, pathfindingResult.length, position)
               );
-
-              const formatedPathSteps: PathStep[] = trainSchedule.path.map((step, i) => ({
-                ...step,
-                coordinates: stepsCoordinates[i],
-                positionOnPath: pathfindingResult.path_items_positions[i],
-              }));
-
               const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
                 operational_points,
                 geometry,
                 pathfindingResult.length
               );
 
-              const allWaypoints = upsertViasInOPs(suggestedOperationalPoints, formatedPathSteps);
+              const updatedPathSteps: PathStep[] = trainSchedule.path.map((step, i) => {
+                const correspondingOp = suggestedOperationalPoints.find(
+                  (suggestedOp) =>
+                    'uic' in step &&
+                    suggestedOp.uic === step.uic &&
+                    suggestedOp.ch === step.secondary_code
```

Comment:
is this condition to find the **correspondingOp** seems good ?

Reply:
When adding new train, it seems to resolve the OD problem, to be confirmed

Reply:
Hm, it sounds a bit weird to me to compare a `ch` with a `secondary_code`, but I'm no train expert. Would someone else be able to chime in and tell if it makes sense?

Reply:
Yes, that makes sense, secondary_code is a bit of a synonym for ch code. But I can't remember why sometimes we use `secondary_code` and other times `ch` code. Perhaps @SharglutDev  could tell us a bit more?

Reply:
it seems to me that's right

Reply:
I think the back called this prop `secondary_code` because we might not always put the ch code in it but another one sometimes ? (but we chosed to use only the ch for now)